source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -104,24 +104,24 @@ absl::optional<int64_t> TimerTokenBucket::remainingFillInterval() const {
                               absl::Seconds((time_after_last_fill) / 1s));
 }
 
-bool TimerTokenBucket::consume(double) {
+bool TimerTokenBucket::consume(double, uint32_t to_consume) {
   // Relaxed consistency is used for all operations because we don't care about ordering, just the
   // final atomic correctness.
   uint32_t expected_tokens = tokens_.load(std::memory_order_relaxed);
   do {
     // expected_tokens is either initialized above or reloaded during the CAS failure below.
-    if (expected_tokens == 0) {
+    if (expected_tokens < to_consume) {
```

Comment:
> I didn't look at this carefully but did you audit everything to make sure there can be no underflow or overflow conditions?

Good point. I am sure underflow or overflow won't happen because we only consum the token when the `expected_tokens` larges then `to_consume`. And they both are uint32_t.

But I think this notices me to add a more strict check to the value range of the `to_consume`.